pkg/image/api/types.go

```diff
@@ -60,6 +84,69 @@ type ImageLayer struct {
 	Size int64
 }
 
+// SignatureState holds a result of signature's server-side check.
+type SignatureState string
+
+// SignatureForImage describes relation of the signature to the image containing it.
+type SignatureForImage string
+
+// ImageSignature holds a signature of an image.
+type ImageSignature struct {
+	// Describes a type of stored blob.
+	Type string
+	// An opaque binary string which is an image's signature.
+	Content []byte
+	// A time of last server-side signature check. IOW a time of the last update of SignatureState.
+	Checked *unversioned.Time
+	// Says whether the signature matches image object containing it.
+	ForImage SignatureForImage
+	// Result of the last server-side signature check performed at Checked. Filled by server.
+	State SignatureState
+
+	// Following metadata fields will be set by server if the signature content is successfully parsed and
+	// the information available.
+
+	// Digest of manifest of the signed image (e.g. "sha256:xxxxx...").
+	ImageHash string
+	// A human readable string representing image's identity. It could contain company's name, a product name
+	// and a version. Eligible value is image's pull specification
+	// (e.g. "registry.access.redhat.com/rhel7/rhel:7.2").
+	ImageIdentity string
+	// Optional. Contains interesting signed claims not important enough to become primary fields.
+	SignedClaims map[string]string
```

Comment:
This will hold arbitrary new claims added in the future that are interesting but not standard enough to become first-class fields. All the claims, once stored, will need to be kept in this mapping as well - even if they later become first-class fields.